core/modules/meet/client/meet-low-level.js

```diff
@@ -1,258 +1,474 @@
+    console.log('DISCONNECT')
 
-    onConnectionSuccess() {
-        log('onConnectionSuccess', arguments)
+    template.connectionStarted = false
+    Meteor.users.update(Meteor.userId(), {
+        $unset: { 'profile.meetRoomName': 1 },
+    })
 
-        meet.room = meet.connection.initJitsiConference(kebabCase(meet.roomName), {})
-        meet.room.on(window.JitsiMeetJS.events.conference.TRACK_ADDED, meet.onTrackAdded)
-        meet.room.on(window.JitsiMeetJS.events.conference.TRACK_REMOVED, meet.onTrackRemoved)
-        meet.room.on(window.JitsiMeetJS.events.conference.CONFERENCE_JOINED, meet.onConferenceJoined)
-        meet.room.on(window.JitsiMeetJS.events.conference.USER_JOINED, () => {
-            log('user join', arguments)
-        })
-        meet.room.on(window.JitsiMeetJS.events.conference.USER_LEFT, meet.onUserLeft)
+    if (template.room?.room) {
+        try {
+            template.room.leave()
+        } catch (err) {
+            console.log('Error during leaving', err)
+        }
+    }
 
-        meet.room.setDisplayName(Meteor.user().profile.name)
+    const connection = template.connection.get()
 
-        meet.room.join()
-    },
+    connection.disconnect()
+    connection.removeEventListener(meetLowLevel.events.connection.CONNECTION_ESTABLISHED, (template) =>
+        onConnectionSuccess(template)
+    )
+    connection.removeEventListener(meetLowLevel.events.connection.CONNECTION_FAILED, onConnectionFailed)
+    connection.removeEventListener(meetLowLevel.events.connection.CONNECTION_DISCONNECTED, () =>
+        onConnectionDisconnected(template)
+    )
 
-    onConnectionFailed() {
-        log('onConnectionFailed', arguments)
-    },
-    onDisconnected() {
-        log('onDisconnected', arguments)
-    },
+    template.room = undefined
+    template.connection.set(undefined)
+}
 
-    onConferenceJoined() {
-        log('onConferenceJoined', arguments)
-        meet.isJoined = true
+/*
+ **  LowMeetJs events listeners
+ */
+const onTrackAdded = (template, track) => {
+    // Since we attach local tracks separately, we do not need attach it again
+    if (track.isLocal()) return
+
+    const participantId = track.getParticipantId()
+    const _remoteTracks = template.remoteTracks.get()
+
+    if (!_remoteTracks[participantId]) _remoteTracks[participantId] = {}
+
+    if (track.getType() === 'video') {
+        // When receiving a 'desktop' track, Jitsi doesn't immediately set the correct type, leading to confusion with our own tracks.
+        // Initially, a 'desktop' track is classified as a 'camera' type, but after a few seconds, it is eventually updated to a 'desktop' track type.
+        // This inconsistency is quite frustrating, and since we haven't found a suitable solution, it's better to introduce a timeout before setting the track type to 'video'.
+        // Cf: https://community.jitsi.org/t/identifying-new-track-as-desktop/118232/2
+
+        setTimeout(() => {
```

Comment:
No,  since we can't differenciate the track on the event 'onTrackAdded', this will replace the actual camera track. Waiting 1 second is not that degrading really, it's like an initialization of the desktop track.